flowmachine/flowmachine/features/subscriber/active_subscribers.py

```diff
@@ -0,0 +1,73 @@
+from datetime import date
+from typing import List, Union, Optional
+from functools import reduce
+
+from flowmachine.core.mixins.exposed_datetime_mixin import ExposedDatetimeMixin
+from flowmachine.core.query import Query
+from flowmachine.features.subscriber.total_active_periods import (
+    TotalActivePeriodsSubscriber,
+)
+from flowmachine.features.utilities.events_tables_union import EventsTablesUnion
+from dateutil.rrule import rrule, DAILY
+
+
+class ActiveSubscribers(ExposedDatetimeMixin, Query):
+    """Returns a list of subscribers seen at least `active_days` between `start_date` and `end_date`,
+    where 'active' is at least `active_hours` call-hours active
+    """
+
+    def __init__(
+        self,
+        start_date: Union[date, str],
+        end_date: Union[date, str],
+        active_hours: int,
+        active_days: int,
+        subscriber_id: str = "msisdn",
+        events_tables: Optional[List[str]] = None,
+        subscriber_subset=None,
+    ):
+        self.start_date = start_date
+        self.end_date = end_date
+        self.active_hours = active_hours
+        self.sub_id_column = subscriber_id
+        self.events_tables = events_tables
+        self.active_days = active_days
+
+        self.events_table_query = EventsTablesUnion(
+            self.start_date,
+            self.end_date,
+            tables=events_tables,
+            subscriber_identifier=subscriber_id,
+            columns=[subscriber_id, "datetime"],
+            subscriber_subset=subscriber_subset,
+        )
+
+        self.hour_queries = [
+            TotalActivePeriodsSubscriber(
+                start=day,
+                total_periods=24,
```

Comment:
Can we expose these in the top level query and not assume that the outer period is a day, and the inner one an hour? Also can we make the hours param available - much more flexible that way.